tests/nested/manual/fakeinstaller-real/task.yaml

```diff
@@ -0,0 +1,243 @@
+  # TODO: XXX: fakeinstaller should work on a empty disk too, it errors
+  # right now with "device with name "/dev/vdc" is not a physical disk
+  echo "label: gpt" | sfdisk "fake-disk.img"
+
+  NESTED_PARAM_EXTRA="-drive file=$(pwd)/fake-disk.img,if=virtio,snapshot=off"
+  export NESTED_PARAM_EXTRA
+
+  # create a VM and mount a cloud image
+  tests.nested build-image classic
+
+  # TODO: nested classic images do not support secure boot today so
+  #       this will not work to test the secure boot installer. So for
+  #       now the workaround is to boot classic to create user/ssh
+  #       keys, shutdown down, convert disk from qcow2->raw and rename
+  #       from classic->core and use nested_start_core_vm (like below)
+  #
+  # start it so that cloud-init creates ssh keys and user
+  nested_start_classic_vm
+
+  # make sure classic image is bootable with snakeoil keys
+  # TODO: move to nested_create_classic_image
+  # XXX: use assets from gadget instead?
+  for s in BOOT/BOOTX64.EFI ubuntu/shimx64.efi; do
+      remote.exec "sudo cp -a /boot/efi/EFI/$s /tmp"
+      remote.exec "sudo chmod 755 /tmp/$(basename $s)"
+      remote.pull /tmp/"$(basename $s)" .
+      nested_secboot_sign_file "$(basename $s)" "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+      remote.push "$(basename $s)"
+      remote.exec "sudo mv $(basename $s) /boot/efi/EFI/$s"
+  done
+
+  # push our snap down
+  # TODO: this abuses /var/lib/snapd to store the deb so that mk-initramfs-classic
+  # can pick it up. the real installer will also need a very recent snapd
+  # in its on disk-image to supprot seeding
+  remote.push "$SPREAD_PATH"/../snapd_*.deb
+  remote.exec "sudo mv snapd_*.deb /var/lib/snapd/"
+  remote.exec "sudo apt install -y /var/lib/snapd/snapd_*.deb"
+    
+  # push our seed down
+  # TODO: merge with classic /var/lib/snapd/seed eventually
+  # XXX: port scp -r to remote.push
+  #remote.push ./classic-seed/system-seed/ '~/'
+  sshpass -p ubuntu scp -r -P 8022 -o ConnectTimeout=10 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no ./classic-seed/system-seed/ user1@localhost:~/install-seed
+  remote.exec "sudo mv /home/user1/install-seed /var/lib/snapd/"
+
+  # XXX: the code in DeviceManager.SystemAndGadgetInfo() will only work on
+  # classic systems with modeenv right now (which is something we may need
+  # to fix to work from the classic installer).
+  # For now pretend we have a modeenv
+  remote.exec 'echo "mode=run" | sudo tee -a /var/lib/snapd/modeenv'
+  remote.exec 'sudo systemctl restart snapd'
+
+  # shutdown the classic vm to install with a core VM that supports
+  # secboot/tpm
+  nested_shutdown
+  sync
+
+  # HACK: convert "classic" qcow2 to raw "core" image because we need
+  # to boot with OVMF we really should fix this so that classic and
+  # core VMs are more similar
+  qemu-img convert -f qcow2 -O raw "$NESTED_IMAGES_DIR/$(nested_get_image_name classic)" "$NESTED_IMAGES_DIR/$(nested_get_image_name core)"
```

Comment:
you can use:
tests.nested get image-name classic
tests.nested get image-name core